### ISSUE TITLE
Make it easier to dynamically update the color of keys

### ISSUE THREAD

**Issue by hallvard** [dlsc-software-consulting-gmbh/KeyboardFX#7]
I'm writing a wordle clone and use keyboardfx for the virtual keyboard. The keyboard control works greit, thanks!

One feature available in the original web ui is colouring the keys based on which letters are known to be wrong, misplaced or correct. I do that by attaching style classes to the keys, but to refresh the skin I need to set the selected keyboard to null then back since this is the only property that triggers rebuilding the underlying grid.

It would be a lot easier if the key's skin could listen to the style classes and immediately update.
Also, it would be nice to have a utility method for finding the key for a specific character/keycode, currently I must traverse alle rows and keys.

**Comment by dlemmermann** [comment]
hmmmmm ..... but normally nodes do immediately restyle when you add or remove a style class ... doesn't that happen for you?

**Comment by dlemmermann** [comment]
For the utility method I would definitely welcome a PR :-)

**Comment by hallvard** [comment]
I have code like this:

```
var kind = charKinds.get(Character.toUpperCase(c.charAt(0)));
if (kind != null) {
     String styleClass = CharSlotNode.getCharSlotKindStyleClass(kind);
     key.getStyleClass().setAll(List.of(styleClass));
} else {
     key.getStyleClass().clear();
}
```

Key/KeyBase isn't a node, so I guess there's no standard logic to trigger restyling. KeyViewBase transfers the style classes of the key with `getStyleClass().addAll(key.getStyleClass())`, but doesn't listen to the observable style class list, though.

**Comment by dlemmermann** [comment]
That's right ... keys are no nodes. Yep, that could be improved. The actual nodes should listen to the keys.

**Comment by hallvard** [comment]
I can see if I can PR that, and perhaps also a Norwegian (mac) keyboard?

**Comment by dlemmermann** [comment]
Sounds good to me :-) But please create issues first and then submit separate PRs.

**Comment by dlemmermann** [comment]
I renamed the ticket to it becomes more obvious there is a bug / issue.

**Comment by dlemmermann** [comment]
Closing due to inactivity.
